pkg/crc/machine/bundle/metadata.go

```diff
@@ -257,7 +257,7 @@ func GetBundleInfoFromName(bundleName string) (*FilenameInfo, error) {
 	var filenameInfo FilenameInfo
 
 	// crc_preset_driver_version_arch_customSuffix.crcbundle
-	bundleNameRegex := regexp.MustCompile(`crc(?:(?:_)([[:alpha:]]+))?_([[:alpha:]]+)_([0-9.]+)_([[:alnum:]]+)(?:(?:_)([0-9]+))?\.crcbundle`)
+	bundleNameRegex := regexp.MustCompile(`crc(?:(?:_)([[:alpha:]]+))?_([[:alpha:]]+)_([[:alnum:].-]+)_([[:alnum:]]+)(?:(?:_)([0-9]+))?\.crcbundle`)
```

Comment:
@redbeam I think we are doing it wrong way since openshift is using semver and as per https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string we should use this regex to make sure it works for every release and we don't have to change it again.

Reply:
wow, that regex is going to be huge :D

Reply:
yes, but makes it more accurate :)

Reply:
If we go with that can you add comment on top of it around how this regex works so that if we want to update it would be each?
```
crc: Matches the fixed crc part.
(?:(?:_)([[:alpha:]]+))?: Matches an optional part with only alphabetic characters (e.g., microshift). The part is optional, so it can be absent.
__([[:alpha:]]+): Matches the next mandatory alphabetic part (e.g., libvirt).
([[:alnum:].-]+): Matches the version, which can include alphanumeric characters, dots (.), and hyphens (-) (e.g., 4.16.7 or 4.16.7-ec.2).
_([[:alnum:]]+): Matches the architecture or platform part (e.g., amd64).
(?:(?:_)([0-9]+))?: Optionally matches a trailing number after an underscore (e.g., 2345).
\.crcbundle: Matches the file extension .crcbundle.
```

Reply:
yeah no problem